### PR TITLE
product domain structure added

### PR DESCRIPTION
**Pull request by tonasanya**

### CODE REVIEW

**Comment by GalaJoe** [comment]
Tabbing in all multi-line comments is 1 indent too far and not identical to the rest of the project.

File: src/domain_factory/product/product.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+            Returns
+            -------
+            List
+                Containing 'record_count' Products
+            """
+
+        records = []
+
+        return records
```
Comment (GalaJoe) [inline src/domain_factory/product/product.py]: Abstract methods do not need an implementation, typically we've used "pass" in these.

Comment (tonasanya) [reply src/domain_factory/product/product.py]: done

**Review by GalateaTom — APPROVED** [review]
Approved

**Pull request merged**